srml/contract/src/vm/mod.rs

```diff
@@ -188,12 +188,20 @@ pub fn execute<'a, E: Ext>(
 		special_trap: None,
 	};
 
-	let mut instance = sandbox::Instance::new(&instrumented_code, &imports, &mut runtime)
-		.map_err(|_| Error::Instantiate)?;
-
-	let run_result = instance.invoke(b"call", &[], &mut runtime);
-
-	to_execution_result(runtime, run_result.err())
+	// Instantiate the instance from the instrumented module code.
+	let exec_error: Option<sandbox::Error> =
+		match sandbox::Instance::new(&instrumented_code, &imports, &mut runtime) {
+			// No errors or traps were generated on instantiation! That
+			// means we can now invoke the contract entrypoint.
+			Ok(mut instance) => instance.invoke(b"call", &[], &mut runtime).err(),
+			// `start` function trapped.
+			Err(err @ sandbox::Error::Execution) => Some(err),
```

Comment:
The issue was in that we performed special error handling (with `to_execution_result`) only for the result of `invoke`. However, that's not correct since `start` function can also execute code and potentially trigger the special case that we are looking for in `to_execution_result`.

So we catch the trap generated by `start` function here, save it in `exec_error` and let 		 	`to_execution_result` do it's job.